### ISSUE TITLE
Slow stepping when debugging Typescript repos

### ISSUE THREAD

**Issue by isidorn** [Microsoft/vscode#23743]
Investigate again why the stepping is slow when debuggin the Typescript repository (steps take 1 second). More details can be found here https://github.com/Microsoft/vscode/issues/5645
fyi @egamma

**Comment by weinand** [comment]
Please try for both protocols.

**Comment by isidorn** [comment]
I have compared the performance while debugging the `checker.ts` file in the typescript repository. Below I have attached three gifs illustrating this. The first one is the `inspector` protocol debugger which shows very good stepping performance (indistinguishable from debugging a simple file). The second shows the `legacy` adapter which is in the current insiders / stable and the third shows the `legacy` adapter with a logging bugfix made by @weinand. When comparing these two it seems like the bugfix did not have a noticable impact on the performance.

Current slowness shown only by the `node` adapter is due the old node debug protocol returning more data than asked for. @weinand would have to dig more in order to find the exect culprit, however that might not be necessery since we are moving towards using the `inspector` protocol in any case.

**User workaround to get a good experience**: add `"protocol": "inspector"` to your `launch.json` or update to node version 8 or greater (since then we will automatically pick the new inspector protocol)

Also created a related issue for case with much more local variables https://github.com/Microsoft/vscode/issues/25296

fyi @egamma 

![checkernode2](https://cloud.githubusercontent.com/assets/1926584/25386787/d1a17d5c-29c8-11e7-86d7-e5ef5888d676.gif)

![checkernode1](https://cloud.githubusercontent.com/assets/1926584/25386755/bbf79504-29c8-11e7-9daf-fab90362850a.gif)

![checkernode1update](https://cloud.githubusercontent.com/assets/1926584/25386929/4f7f9ca4-29c9-11e7-905e-fe2a928ccf75.gif)
